crates/transaction-pool/src/traits.rs

```diff
@@ -272,6 +275,118 @@ pub trait PoolTransaction: fmt::Debug + Send + Sync + FromRecoveredTransaction {
     fn size(&self) -> usize;
 }
 
+// #[main_codec]
+#[derive(Debug, Clone, PartialEq, Eq, Hash)]
+pub(crate) struct PooledTransaction {
+    /// EcRecovered transaction info
+    pub(crate) transaction: TransactionSignedEcRecovered,
+
+    /// For EIP-1559 transactions that is `feeCap x gasLimit + transferred_value
+    pub(crate) cost: U256,
+
+    /// This is `priority + basefee`for EIP-1559 and `gasPrice` for legacy transactions.
+    pub(crate) effective_gas_price: U256,
+
+    /// Measurement of the heap usage of this type and all its internals.
+    pub(crate) size: usize,
```

Comment:
we don't need this if we're just using the len of the input.

we need to keep this type as small as possible